bootstrap/src/stream.ml

```diff
@@ -0,0 +1,576 @@
+open Rudiments
+
+type 'a elm =
+  | Nil
+  | Cons of 'a * 'a t
+and 'a t = 'a elm lazy_t
+
+let empty = lazy Nil
+
+let init n ~f =
+  let rec fn i n f = lazy begin
+    match i < n with
+    | false -> Nil
+    | true -> begin
+      let elm = f i in
+      let t = fn (Usize.succ i) n f in
```

Comment:
I think `(Usize.succ i)` can be shortened to `(succ i)` here.  Only in modules where we implement `succ` do we have to be explicit about `Usize`.